File: src/main/java/com/braintreegateway/SubscriptionTransactionRequest.java
package com.braintreegateway;

import java.math.BigDecimal;

public class SubscriptionTransactionRequest extends BaseRequest {

    private BigDecimal amount;
    private String subscriptionId;

    public SubscriptionTransactionRequest amount(BigDecimal amount) {
        this.amount = amount;
        return this;
    }

    public SubscriptionTransactionRequest subscriptionId(String subscriptionId) {
        this.subscriptionId = subscriptionId;
        return this;
    }
    
    @Override
    public String toXmlString() {
        return buildRequest("transaction").toXmlString();
    }

    protected RequestBuilder buildRequest(String root) {
        return new RequestBuilder(root).
            addElement("amount", amount).
            addElement("subscriptionId", subscriptionId).
            addElement("type", Transactions.Type.SALE.toString().toLowerCase());
    }
}

File: src/main/scala/com/braintreegateway/SubscriptionTransactionRequest.scala
package com.braintreegateway

import scala.math.BigDecimal

class SubscriptionTransactionRequest extends BaseRequest {

  private var amount: Option[BigDecimal] = None
  private var subscriptionId: Option[String] = None

  def amount(amount: BigDecimal): SubscriptionTransactionRequest = {
    this.amount = Option(amount)
    this
  }

  def subscriptionId(subscriptionId: String): SubscriptionTransactionRequest = {
    this.subscriptionId = Option(subscriptionId)
    this
  }

  override def toXmlString: String = {
    buildRequest("transaction").toXmlString
  }

  protected def buildRequest(root: String): RequestBuilder = {
    new RequestBuilder(root).
      addElement("amount", amount).
      addElement("subscriptionId", subscriptionId).
      addElement("type", Transactions.Type.SALE.toString.toLowerCase)
  }
}